bundle/config/mutator/load_git_details.go

```diff
@@ -33,28 +34,56 @@ func (m *loadGitDetails) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagn
 		b.WorktreeRoot = vfs.MustNew(info.WorktreeRoot)
 	}
 
-	b.Config.Bundle.Git.ActualBranch = info.CurrentBranch
-	if b.Config.Bundle.Git.Branch == "" {
-		// Only load branch if there's no user defined value
-		b.Config.Bundle.Git.Inferred = true
-		b.Config.Bundle.Git.Branch = info.CurrentBranch
-	}
+	config := &b.Config.Bundle.Git
 
-	// load commit hash if undefined
-	if b.Config.Bundle.Git.Commit == "" {
-		b.Config.Bundle.Git.Commit = info.LatestCommit
+	config.ActualBranch = info.CurrentBranch
+	if config.Branch == "" {
+		config.Inferred = true
 	}
 
-	// load origin url if undefined
-	if b.Config.Bundle.Git.OriginURL == "" {
-		b.Config.Bundle.Git.OriginURL = info.OriginURL
-	}
+	// The value from config takes precedence; however, we always warn if configValue and fetchedValue disagree.
+	// In case of branch and commit and absence of --force we raise severity to Error
+	diags = append(diags, checkMatch("Git branch", info.CurrentBranch, &config.Branch, b.Config.Bundle.Force)...)
+	diags = append(diags, checkMatch("Git commit", info.LatestCommit, &config.Commit, b.Config.Bundle.Force)...)
+	diags = append(diags, checkMatch("Git originURL", info.OriginURL, &config.OriginURL, true)...)
 
 	relBundlePath, err := filepath.Rel(b.WorktreeRoot.Native(), b.BundleRoot.Native())
 	if err != nil {
 		diags = append(diags, diag.FromErr(err)...)
 	} else {
 		b.Config.Bundle.Git.BundleRootPath = filepath.ToSlash(relBundlePath)
 	}
+
+	config.BundleRootPath = filepath.ToSlash(relBundlePath)
```

Comment:
This looks identical to the `else` branch above. Intentional?

Reply:
No, good catch.